### PR TITLE
doc(kitex): translate kitex custom template doc

### PR DESCRIPTION
**Pull request by cqqqq777**
#### What type of PR is this?

docs
#### What this PR does / why we need it (en: English/zh: Chinese):

en: translate kitex custom template doc based on Chinese doc
zh:

### CODE REVIEW

**Comment by cqqqq777** [comment]
[this issue](https://github.com/cloudwego/kitex/issues/856)

**Comment by GuangmingLuo** [comment]
@cqqqq777 thanks for your contribution, please try to fix typo and punctuation problems first.

**Comment by cqqqq777** [comment]
sorry,I've fixed it

File: content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md [hunk 1]
```diff
@@ -1,6 +1,131 @@
 ---
 title: "Custom Scaffold Template"
-date: 2023-03-10
+date: 2023-03-31
 weight: 7
 description: >
 ---
+
+Kitex provides the ability to customize templates,if the default templates can not meet your needs,you can use the function of this custom template provided by Kitex. Now ,it supports rendering single file and circular rendering according to methodInfo
```
Comment (GuangmingLuo) [inline content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md:8]: 第一段三个逗号的空格都有问题，有的缺少空格，有的空格用错了

Comment (GuangmingLuo) [reply content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md:8]: 逗号后面接一个空格再开始下一句，也检查一下下面的段落

**Comment by cqqqq777** [comment]
不好意思这是我第一次翻译文档，现在我又检查了一遍，你看一下可以吗

File: content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md [hunk 2]
```diff
@@ -1,6 +1,131 @@
...
+
+
+
+# How to Use
+
+1. The data used by the template is PackageInfo and we assume that this section contains all the metadata(e.g. methodInfo ...). You only needs to pass the template file, and the data in the template is PackageInfo data.
+
+2. Template rendering can only render a single file at a time. If there is a situation of sorting the file by method and rendering it, you need to control them in your code.
+
+3. Template files are passed through the YAML folder and are specified with the `-template-dir` command-line argument. Because writing all the templates to a file causes the file to be huge, so specify a folder instead.
```
Comment (Duslia) [inline content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md:18]: Please improve this translation. It is chinglish

File: content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md [hunk 3]
```diff
@@ -1,6 +1,131 @@
...
+
+4. `extensions.yaml` in the YAML folder is a specific file. The content of this configuration file is [extending the service code](https://www.cloudwego.io/docs/kitex/tutorials/code-gen/template_extension/). If it exists, you do not need to pass the 'template-extension' parameter.
```
Comment (Duslia) [inline content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md:20]: You do not need -> there is no need

File: content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md [hunk 4]
```diff
@@ -1,6 +1,131 @@
...
+
+5. When updating, currently only three options are supported: overwrite, skip, and add files based on methods, appending in a file is also supported. Adding files means that if the user adds new methods while rendering circularly according to methods, new files will be created.
```
Comment (Duslia) [inline content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md:22]: Please improve this translation. It is chinglish

File: content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md [hunk 5]
```diff
@@ -1,6 +1,131 @@
...
+
+6. The code generated by Kitex is divided into two parts, 'kitex_gen' and 'mainPkg' (including 'main.go', 'handler.go' ...). 'kitex_gen' is immutable but you can only choose one between 'mainPkg' and 'custom layout'. If  'custom layout'  is specified, 'mainPkg' will not be generated again.
```
Comment (Duslia) [inline content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md:24]: Please separate the sentence into two.

File: content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md [hunk 6]
```diff
@@ -1,6 +1,131 @@
...
+date: 2023-04-03
 weight: 7
 description: >
 ---
+
+Kitex provides the ability to customize templates, if the default templates can not meet your needs, you can use the function of this custom template provided by Kitex. Now, it supports rendering single file and circular rendering according to methodInfo.
+
+
+
+# How to Use
```
Comment (GuangmingLuo) [inline content/en/docs/kitex/Tutorials/code-gen/custom_tpl.md:12]: 正文中起始用二级标题 ##

**Comment by cqqqq777** [comment]
ok

**Comment by GuangmingLuo** [comment]
@cqqqq777 Please adapt this pr, thanks: https://github.com/cloudwego/cloudwego.github.io/pull/597

**Review by GuangmingLuo — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

**Pull request merged**